documentation/specs/proposed/BuildCheck.md

```diff
@@ -0,0 +1,405 @@
+
+Non-Goals (but might be considered):
+* bulk configuration of multiple rules - based on analyzers/rules prefixes or/and categories.
+* attempts to try to configure standard msbuild warnings/errors via `.editorconfig` should lead to fail fast errors.
+* configuring analysis levels when analysing from binlog - beyond the collected editorconfigs.
+* Aliasing the analyzers/rules, allowing to create multiple instances with different custom configuration (e.g. single analyzer checking configurable list of forbidden properties prefixes can have 2 instance, each initialized with different list to check, each of the instance configurable for individual projects separately).
+
+Out of scope for configuration:
+* opt-out of analysis on code-level (analogy to C# pragmas, but within msbuild xml files).
+* lower granularity of `.editorconfig` settings other than whole projects.
+* attributing configuration to a .sln file and expecting it will apply to all contained projects.
+* Support for multiple [custom configurations](#custom-configuration-declaration) within a single build for a single rule. (Not to be mixed with [standardized configuration](#standardized-configuration-declaration) - which can be configured freely per project) If a custom configuration will be used, it will need to be specified identically in each explicit configurations of the rule. This is chosen so that there are no implicit requirements on lifetime of the analyzer or analyzer instancing – each analyzer will be instantiated only once per build (this is however something that will very likely change in future versions – so authors are advised not to take hard dependency on single instance policy).
+
+### User Configurable Options
+
+Initial version of BuildCheck plans a limited set of options configurable by user (via `.editorconfig`) by which users can override default configuration of individual analyzer rules.
+
+#### Enablement
+
+Boolean option `IsEnabled` will be available to allow users to disable/enable particualr rule.
+
+Different rules of a single analyzer can have different enabledment status configured.
+
+If all the rules from a single analyzer are disabled - analyzer won't be given any data for such configured part of the build (specific project or a whoe build). If analyzer have some rules enabled and some disabled - it will be still fed with data, but the reports will be post-filtered.
+
+Same rule can have different enablement status for different projects.
+
+#### Severity
+
+Option `Severity` with following values will be available:
+
+* `Message`
+* `Warning`
+* `Error`
+
+Configuration will dictate transformation of the analyzer report to particular build output type (message, warning or error).
+
+Different rules of a single analyzer can have different severities configured. Same rule can have different severities for different projects.
+
+#### Scope of Analysis
+
+Option `EvaluationAnalysisScope` with following possible options will be avaialable:
```

Comment:
What's the right level of configuration here? I was thinking of this as something that the analyzer/rule declared, rather than something end-user configurable. But I'm not sure that's right!

Reply:
It's both - analyzer provides defualt, user can override.

If during the authoring we find out there might be rules where user-overriding is not desired - let's then think about selectively preventing